smac_planner/src/a_star.cpp

```diff
@@ -313,6 +321,32 @@ bool AStarAlgorithm<NodeT>::createPath(
     // 2) Mark Nbest as visited
     current_node->visited();
 
+    if (_motion_model == MotionModel::DUBIN || _motion_model == MotionModel::REEDS_SHEPP) {
+      // This must be a NodeSE2 node if we are using these motion models
+
+      // 2.a) Use an analytic expansion (if available) to generate a path
+      // to the goal.
+      // Then check if it is collision free.
+      // TODO(james-ward): Don't always run this check - base it on the
+      // number of iterations and how close to the goal we are
+      // Always run the expansion on the first run in case there is a
+      // trivial path to be found
+      if (analytic_iterations <= 0) {
```

Comment:
How often does this now work? Last iteration of this logic never actually ran until the very end of the path and never in the middle (or periodically with increasing frequency the closer to the goal).

Reply:
In the TB3 world this would run around 4-5 times for a path on the opposite side with obstacles obstructing it.